src/window/mapped.rs

```diff
@@ -183,6 +183,27 @@ impl MappedId {
     pub fn get(self) -> u64 {
         self.0
     }
+
+    /// Converts the ID to a string that can be used as an identifier in
+    /// ext_foreign_toplevel_handle_v1::identifier
+    ///
+    /// > An identifier is a string that contains up to 32 printable ASCII bytes. An identifier must
+    /// > not be an empty string.
+    ///
+    /// Since the ID is exposed to IPC, it's useful for this conversion to be stable and reversible.
+    /// That way, clients can associate a foreign toplevel handle with an IPC window ID.
+    ///
+    /// We use the decimal representation of the ID, which is up to 20 characters long for u64::MAX.
+    /// This is within the 32-character limit, and is nice because it matches up with how `niri msg`
+    /// prints the IDs to the console.
+    ///
+    /// We also prepend "mapped-" prefix in case we want to extend this namespace in the future.
+    /// (Xwayland maybe? it'd have the same ID counter, but i figure it's better to futureproof it).
+    ///
+    /// This brings the total length to 27 characters for u64::MAX.
+    pub fn to_protocol_string(self) -> String {
+        format!("mapped-{}", self.0)
```

Comment:
Realistically, i think this `mapped-` prefix adds nothing: that is only because it derives from a struct called `MappedId`, but that's kind of an internal naming? We could get away with just the numeric ID as a string here: if we need other namespaces for any reason, adding a prefix *is* fully backwards compatible. But when would we ever need another namespace?

Reply:
I've gone ahead and changed it to just be the ID string, without a prefix.

Like i said, we can add prefixes in the future (or a "don't parse" me random bytes implementation also cannot ever overlap, if it is fixed 32 length).